cmds/core/head/head.go

```diff
@@ -0,0 +1,82 @@
+// Copyright 2024 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package main
+
+import (
+	"bufio"
+	"flag"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"path/filepath"
+)
+
+var errCombine = fmt.Errorf("can't combine line and byte counts")
+
+func run(stdout io.Writer, stderr io.Writer, bytes, count int, files ...string) error {
+	if bytes > 0 && count > 0 {
+		return errCombine
+	}
+
+	var printBytes bool
+	var buffer []byte
+	if bytes > 0 {
+		printBytes = true
+		buffer = make([]byte, bytes)
+	}
+
+	if count == 0 {
+		count = 10
+	}
+
+	var newLineHeader bool
+
+	for _, file := range files {
+		f, err := os.Open(file)
+		if err != nil {
+			fmt.Fprintf(stderr, "head: %v", err)
+			continue
+		}
+		if len(files) > 1 {
+			if newLineHeader {
+				fmt.Fprintf(stdout, "\n==> %s <==\n", filepath.Base(f.Name()))
+			} else {
+				fmt.Fprintf(stdout, "==> %s <==\n", filepath.Base(f.Name()))
+				newLineHeader = true
+			}
+		}
+		if printBytes {
+			n, err := f.Read(buffer)
```

Comment:
I wonder if we should use ReadFull here. On plan 9, sockets and pipes are named files, and so could be passed to head. Do we want to block on the file or return what is there at the moment this command is run?

Reply:
I'm wondering if you could just wrap the file with a LimitedReader, then just io.Copy from that limited reader to stdout. This would avoid the fooling around with byte slices and make?

Reply:
For limit reader we can use CopyN, it will make source file smaller for sure. But it will create 4K buffer every time, so I think this tiny slice make some sense.

As for io.ReadFull I don't see issue here, because it will read exactly -c bytes. I will add it.